libstuff/SLog.cpp

```diff
@@ -50,7 +58,11 @@ string addLogParams(string&& message, const map<string, string>& params) {
             message += " ";
         }
         const auto& param = *next(params.begin(), i);
-        message += param.first + ": '" + param.second + "'";
+        string value = param.second;
```

Comment:
No need to copy this, just append param.second below.

Reply:
> append param.second below

What do you mean? It's const, so I don't understand

Reply:
I think you could simplify this code to:

```cpp
    message += " ~~";
    for (const auto& [key, value] : params) {
        message += " " + key + ": '" + (SContains(PARAMS_WHITELIST, key) ? value : "<REDACTED>") + "'";
    }
```

Notice that we append `" ~~"` and not `" ~~ "` so you don't have to do the `index > 0` special handling.

Reply:
IMO this is hard(er) to read

Reply:
Maybe... but the `for (const auto& [key, value] : params) ` is definitely nicer or you don't like those?